src/device-state.coffee

```diff
@@ -0,0 +1,512 @@
+Promise = require 'bluebird'
+_ = require 'lodash'
+Lock = require 'rwlock'
+EventEmitter = require 'events'
+fs = Promise.promisifyAll(require('fs'))
+express = require 'express'
+bodyParser = require 'body-parser'
+
+network = require './network'
+
+constants = require './lib/constants'
+validation = require './lib/validation'
+device = require './lib/device'
+updateLock = require './lib/update-lock'
+migration = require './lib/migration'
+
+DeviceConfig = require './device-config'
+Logger = require './logger'
+ApplicationManager = require './application-manager'
+
+validateLocalState = (state) ->
+	if state.name? and !validation.isValidShortText(state.name)
+		throw new Error('Invalid device name')
+	if state.apps? and !validation.isValidAppsObject(state.apps)
+		throw new Error('Invalid apps')
+	if state.config? and !validation.isValidEnv(state.config)
+		throw new Error('Invalid device configuration')
+
+validateDependentState = (state) ->
+	if state.apps? and !validation.isValidDependentAppsObject(state.apps)
+		throw new Error('Invalid dependent apps')
+	if state.devices? and !validation.isValidDependentDevicesObject(state.devices)
+		throw new Error('Invalid dependent devices')
+
+validateState = Promise.method (state) ->
+	throw new Error('State must be an object') if !_.isObject(state)
+	validateLocalState(state.local) if state.local?
+	validateDependentState(state.dependent) if state.dependent?
+
+class DeviceStateRouter
+	constructor: (@deviceState) ->
+		{ @applications } = @deviceState
+		@router = express.Router()
+		@router.use(bodyParser.urlencoded(extended: true))
+		@router.use(bodyParser.json())
+
+		@router.post '/v1/reboot', (req, res) =>
+			force = validation.checkTruthy(req.body.force)
+			@deviceState.executeStepAction({ action: 'reboot' }, { force })
+			.then (response) ->
+				res.status(202).json(response)
+			.catch (err) ->
+				if err instanceof updateLock.UpdatesLockedError
+					status = 423
+				else
+					status = 500
+				res.status(status).json({ Data: '', Error: err?.message or err or 'Unknown error' })
```

Comment:
Why does the return object look like Go serialised it? (i.e `Data` instead of `data`)

Reply:
Cause in non-error cases the response comes from gosuper that uses that format, and this is a legacy endpoint that has to respect that format (as much as possible, at least).
v2 endpoints will not have this problem.